subsys/bluetooth/controller/hci/hci.c

```diff
@@ -941,6 +941,25 @@ static void le_set_adv_param(struct net_buf *buf, struct net_buf **evt)
 
 	min_interval = sys_le16_to_cpu(cmd->min_interval);
 
+	if (IS_ENABLED(CONFIG_BT_CTLR_PARAM_CHECK) &&
+	    (cmd->type != BT_LE_ADV_DIRECT_IND)) {
+		u16_t max_interval = sys_le16_to_cpu(cmd->max_interval);
+
+		if ((min_interval > max_interval) ||
+		    (min_interval < 0x0020) ||
+		    (max_interval > 0x4000)) {
+			status = BT_HCI_ERR_INVALID_PARAM;
+
+			goto le_set_adv_param_exit;
+		}
+
+		if (min_interval == max_interval) {
```

Comment:
Given that the code below passes only `min_interval`, I believe this should check for inequality.

Reply:
This particular check is not covered on the host side and should probably be always active.

Reply:
From `subsys/bluetooth/mesh/adv.c` `	param.interval_max = param.interval_min;`
This seems like a breaking change. Is this really a pass criteria?

The spec says:
```
The Advertising_Interval_Min shall be less than or equal to the
Advertising_Interval_Max. The Advertising_Interval_Min and
Advertising_Interval_Max should not be the same value to enable the
Controller to determine the best advertising interval given other activities.
```
So there is a shall on `min < max`, but a should on `min == max` so I don't think it is invalid. But recommended to give a range.
For mesh it is is required to set a specific interval because mesh needs control over packet transmission, so I think we need to allow that.
@trond-snekvik Anything you wish to add?

Reply:
Agreeing with @joerchan, if equality doesn't have to be enforced, it shouldn't be, as it would break existing functionality.

Reply:
There is TS errata 12954 to remove the second pass criteria, hence removing in this PR too.